lib/backend/k8s/resources/node.go

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/projectcalico/libcalico-go/lib/backend/api"
+ 	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	"github.com/projectcalico/libcalico-go/lib/errors"
+
+	"k8s.io/client-go/kubernetes"
+	kapiv1 "k8s.io/client-go/pkg/api/v1"
+	metav1 "k8s.io/client-go/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+)
+
+func NewNodeClient(c *kubernetes.Clientset, r *rest.RESTClient) api.Client {
+	return &nodeClient{
+		clientSet: c,
+		tprClient: r,
+	}
+}
+
+// Implements the api.Client interface for Nodes.
+type nodeClient struct{
+	clientSet *kubernetes.Clientset
+	tprClient *rest.RESTClient
+}
+
+func (c *nodeClient) Create(kvp *model.KVPair) (*model.KVPair, error){
+	log.Warn("Operation Create no supported on Node type")
+	return nil, errors.ErrorOperationNotSupported{
+		Identifier: kvp.Key,
+		Operation: "Create",
+	}
+}
+
+func (c *nodeClient) Update(kvp *model.KVPair) (*model.KVPair, error){
+	log.Warn("Operation Update no supported on Node type")
+	return nil, errors.ErrorOperationNotSupported{
+		Identifier: kvp.Key,
+		Operation: "Update",
+	}
+}
+
+func (c *nodeClient) Apply(kvp *model.KVPair) (*model.KVPair, error){
+	log.Warn("Operation Apply no supported on Node type")
+	return nil, errors.ErrorOperationNotSupported{
+		Identifier: kvp.Key,
+		Operation: "Apply",
+	}
+}
+
+func (c *nodeClient) Delete(kvp *model.KVPair) error {
+	log.Warn("Operation Delete no supported on Node type")
+	return errors.ErrorOperationNotSupported{
+		Identifier: kvp.Key,
+		Operation: "Delete",
+	}
+}
+
+func (c *nodeClient) Get(key model.Key) (*model.KVPair, error){
+	log.Debug("Received Get request on Node type")
+	node, err := c.clientSet.Nodes().Get(key.(model.NodeKey).Hostname, metav1.GetOptions{})
+	if err != nil {
+		return nil, K8sErrorToCalico(err, key)
+	}
+
+	cn := K8sNodeToCalico(node)
+
+	kvp := &model.KVPair{
```

Comment:
Hm, K8sNodeToCalico() already returns a *model.KVPair.  Why do you need to re-create it here?